common/src/main/java/org/apache/sedona/common/utils/RasterUtils.java

```diff
@@ -467,10 +466,11 @@ public static AffineTransform2D translateAffineTransform(
 
   public static Point2D getWorldCornerCoordinates(GridCoverage2D raster, int colX, int rowY)
       throws TransformException {
+    Point2D.Double gridCoord = new DirectPosition2D(colX - 1, rowY - 1);
     return raster
         .getGridGeometry()
         .getGridToCRS2D(PixelOrientation.UPPER_LEFT)
-        .transform(new GridCoordinates2D(colX - 1, rowY - 1), null);
+        .transform(gridCoord, null);
```

Comment:
`GridCoordinates2D` is not a subclass of `Point2D.Double`, so the result of transform will be `Point2D.Float`: https://github.com/openjdk/jdk/blob/jdk-17%2B35/src/java.desktop/share/classes/java/awt/geom/AffineTransform.java#L2899-L2906. We explicitly use `Point2D.Double` here to avoid loss of precision.